### PR TITLE
feat: added multiselect dropdown

### PR DESCRIPTION
**Pull request by AbdulTheActivePiecer**
## What does this PR do?

Makes a multiselect dropdown available as a property for pieces

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?
Have an action within a piece that's of this type, make sure refreshers work and values are saved as an array.

### CODE REVIEW

**Comment by AbdulTheActivePiecer** [comment]
@abuaboud I can't get the multipleDropdown property value to be of type array it is always a single value, it takes the value's type from the dropdown state I think.

**Comment by AbdulTheActivePiecer** [comment]
also for some reason changes made on pieces don't get detected unless I restart the server by exiting the terminal and re running nx serve backend

File: packages/pieces/framework/src/lib/framework/property/dropdown-prop.ts [hunk 1]
```diff
@@ -14,11 +14,16 @@ export type DropdownOption<T> = {
 	value: T;
 };
 
-export type DropdownProperty<T, R extends boolean> =  BasePropertySchema & {
+export type DropdownProperty<T, R extends boolean> = BasePropertySchema & {
 	refreshers: string[];
-	options: (propsValue: Record<string, OAuth2PropertyValue | number | string | DropdownState<any> | BasicAuthPropertyValue>) => Promise<DropdownState<T>>
+	options: (propsValue: Record<string, OAuth2PropertyValue | number | string | object | BasicAuthPropertyValue | any[]>) => Promise<DropdownState<T>>
```
Comment (abuaboud) [inline packages/pieces/framework/src/lib/framework/property/dropdown-prop.ts]: Revert back DropdownState<any> changes, and You need MultiDropdownState<any>

File: packages/pieces/framework/src/lib/framework/property/dropdown-prop.ts [hunk 2]
```diff
@@ -14,11 +14,16 @@ export type DropdownOption<T> = {
...
 } & TPropertyValue<T, PropertyType.DROPDOWN, R>;
 
 export type StaticDropdownProperty<T, R extends boolean> = BasePropertySchema & {
 	options: DropdownState<T>;
 } & TPropertyValue<T, PropertyType.STATIC_DROPDOWN, R>;
+
+export type MultipleDropdownProperty<T, R extends boolean> = BasePropertySchema & {
+	refreshers: string[];
+	options: (propsValue: Record<string, OAuth2PropertyValue | number | string | object | BasicAuthPropertyValue | any[]>) => Promise<DropdownState<T>>
```
Comment (abuaboud) [inline packages/pieces/framework/src/lib/framework/property/dropdown-prop.ts]: You need to create DropdownState<T> since it's return different type

Comment (abuaboud) [inline packages/pieces/framework/src/lib/framework/property/dropdown-prop.ts]: Remove object and any[] and replace it with right props

**Pull request merged**